unittests/CMakeLists.txt

```diff
@@ -120,7 +120,30 @@ if (CUSTATEVEC_ROOT AND CUDA_FOUND)
     cudaq-platform-mqpu
     nvqir-custatevec-fp64
     gtest_main)
-    gtest_discover_tests(test_mqpu PROPERTIES LABELS "gpu_required")
+  gtest_discover_tests(test_mqpu PROPERTIES LABELS "gpu_required")
+
+  # Test CUDAQ_OBSERVE_FROM_SAMPLING=ON mode.
```

Comment:
Just to confirm ... we also have tests w/ CUDAQ_OBSERVE_FROM_SAMPLING disabled, right? I assume all the prior custatevec tests will effectively be running with CUDAQ_OBSERVE_FROM_SAMPLING effectively off.

Reply:
yes, that's right. All the prior tests (using `observe` without shots) should be running in the `CUDAQ_OBSERVE_FROM_SAMPLING=OFF`  (new code path using batched observe).